source/fundamentals/aggregation/aggregation-expression-operations.txt

```diff
@@ -438,50 +474,127 @@ types using the methods described in this section.
+this section.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 40 60
+
+   * - Java Method
+     - Aggregation Pipeline Operator
+
+   * - `cond() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlBoolean.html#cond(T,T)>`__
+     - :manual:`$cond </reference/operator/aggregation/cond/>`
+
+   * - `switchOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#switchOn(java.util.function.Function)>`__
+
+       | **Typed Variants**:
+       | `switchArrayOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#switchArrayOn(java.util.function.Function)>`__
+       | `switchBooleanOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlBoolean.html#switchBooleanOn(java.util.function.Function)>`__
+       | `switchDateOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDate.html#switchDateOn(java.util.function.Function)>`__
+       | `switchDocumentOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#switchDocumentOn(java.util.function.Function)>`__
+       | `switchIntegerOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlInteger.html#switchIntegerOn(java.util.function.Function)>`__
+       | `switchMapOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#switchMapOn(java.util.function.Function)>`__
+       | `switchNumberOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlNumber.html#switchNumberOn(java.util.function.Function)>`__
+       | `switchStringOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#switchStringOn(java.util.function.Function)>`__
+
+     - :manual:`$switch </reference/operator/aggregation/switch/>`
 
 .. include:: /includes/fundamentals/agg-mql-diff.rst
 
-Suppose you want to have a collection of student data that includes
-their graduation years, which are stored as strings. You want to
-calculate the year of their five-year reunion and store this value in a
-new field.
+Suppose you have a collection of customers with their membership information.
+Originally, customers were either members or not. Over time, membership levels
+were introduced and used the same field. The information stored in this field
+can be one of a few different types, and you want to create a consistent field
+indicating their membership level.
 
-To convert the ``graduationYear`` field to an integer,
-use the ``parseInteger()`` method. To add five years to this value, use the
-``add()`` method. To store the result as ``reunionYear``,
-use the ``addFields()`` method.
+To change what you are doing based on the data type of ``member``, use the
+``switchOn()`` method. If the data type is a string, it indicates the
+membership level and you want to use the data without altering it. If the data
+type is ``Boolean``, check the value with the ``cond()`` method and return a
```

Comment:
Question:
This reads as tutorial instructions rather than a description of the logic. Would it be possible to describe as a description of the logic instead?

Reply:
I tried to pull the descriptions into the previous paragraph and make this paragraph modeled like the others in this topic. I'm not sure if this addresses your concerns, but I think it now models what the other sections have.